source/index.ts

```diff
@@ -0,0 +1,746 @@
+import fs = require('fs');
+import path = require('path');
+import crypto = require('crypto');
+import assert = require('assert');
+import EventEmitter = require('events');
+import dotProp = require('dot-prop');
+import makeDir = require('make-dir');
+import pkgUp = require('pkg-up');
+import envPaths = require('env-paths');
+import writeFileAtomic = require('write-file-atomic');
+import Ajv = require('ajv');
+import debounceFn = require('debounce-fn');
+import semver = require('semver');
+import onetime from 'onetime';
+
+const plainObject: () => object = () => Object.create(null);
+const encryptionAlgorithm = 'aes-256-cbc';
+
+// Prevent caching of this module so module.parent is always accurate
+delete require.cache[__filename];
+const parentDir = path.dirname((module.parent && module.parent.filename) || '.');
+
+const checkValueType = (key: string, value: unknown): void => {
+	const nonJsonTypes = [
+		'undefined',
+		'symbol',
+		'function'
+	];
+
+	const type = typeof value;
+
+	if (nonJsonTypes.includes(type)) {
+		throw new TypeError(`Setting a value of type \`${type}\` for key \`${key}\` is not allowed as it's not supported by JSON`);
+	}
+};
+
+const INTERNAL_KEY = '__internal__';
+const MIGRATION_KEY = `${INTERNAL_KEY}.migrations.version`;
+
+export default class Conf<T = any> implements Iterable<[string, T]> {
+	_options: Options;
+
+	_defaultValues: DefaultValues = {};
+
+	_validator?: Ajv.ValidateFunction;
```

Comment:
This should not be optional.

Reply:
In the codebase the validator only exists if there's a schema. Doesn't that make it optional?

Reply:
@sindresorhus Looks correct to me as well. It will only exist if `options.schema` is present. The validate method already check its existence.